Observer/ProductSaveAfter.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+use Klaviyo\Reclaim\Helper\Logger;
+use Klaviyo\Reclaim\Model\ProductsFactory;
+
+use Magento\Catalog\Model\CategoryFactory;
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class ProductSaveAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * Klaviyo logger helper
+     * @var \Klaviyo\Reclaim\Helper\Logger $klaviyoLogger
+     */
+    protected $_klaviyoLogger;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected $_webhookHelper;
+    protected $_categoryFactory;
+    protected $_klProductFactory;
+    protected $_stockRegistry;
+    protected $product_category_names = [];
```

Comment:
We could remove a few unused protected variables and add variable descriptions to each